packages/ui-extensions-react/src/surfaces/admin/render.tsx

```diff
@@ -25,18 +25,22 @@ import {ExtensionApiContext} from './context';
  */
 export function reactExtension<ExtensionPoint extends RenderExtensionPoint>(
   target: ExtensionPoint,
-  render: (api: ApiForRenderExtension<ExtensionPoint>) => ReactElement<any>,
+  render: (
+    api: ApiForRenderExtension<ExtensionPoint>,
+  ) => ReactElement<any> | Promise<ReactElement<any>>,
 ): ExtensionPoints[ExtensionPoint] {
   // TypeScript can’t infer the type of the callback because it’s a big union
   // type. To get around it, we’ll just fake like we are rendering the
   // Playground extension, since all render extensions have the same general
   // shape (`RenderExtension`).
-  return extension<'Playground'>(target as any, (root, api) => {
-    return new Promise((resolve, reject) => {
+  return extension<'Playground'>(target as any, async (root, api) => {
+    const element = await render(api as ApiForRenderExtension<ExtensionPoint>);
```

Comment:
Shouldn't that be in the `try/catch`? If `render` throws we wouldn't reach the console.error, would we?

Reply:
Good call, I will move the try/catch around the whole function body.

Reply:
err actually @marvinhagemeister do you know if the `console.error` is even needed anymore? I feel like we should be doing that in the sandbox instead.